app/scripts/views/settings/avatar_camera.js

```diff
@@ -119,17 +119,16 @@ function (_, Cocktail, canvasToBlob, FormView, ProgressIndicator,
       this._avatarProgressIndicator.start(this.$('.progress-container'));
 
       this.canvas = this.$('#canvas')[0];
-      this.width = 320;
-      this.height = 0;
+      this.width = 0;
+      this.height = this.displayLength;
 
       this.video[0].addEventListener('canplay', _.bind(this.canPlay, this), false);
     },
 
     canPlay: function () {
       if (! this.streaming) {
+        this.width = this.video[0].videoWidth / (this.video[0].videoHeight / this.height);
```

Comment:
If `this.height` is 0, or `this.video[0].videoHeight` is 0, there will be a divide by 0 error. Should this be guarded against?